Extension/src/Debugger/configurationProvider.ts

```diff
@@ -90,30 +92,20 @@ abstract class DefaultConfigurationProvider implements IConfigurationAssetProvid
         return initialConfigurations;
     }
 
-    // TODO: Update this function when VsCode enables an API for configuration snippet providers
-    public getConfigurationSnippets(): string {
-        if (util.packageJson.contributes.debuggers[0] && !util.packageJson.contributes.debuggers[0].configurationSnippets) {
+    public getConfigurationSnippets(): vscode.CompletionItem[] {
+        let completionItems: vscode.CompletionItem[] = [];
             let configurationSnippet: IConfigurationSnippet[] = [];
 
             this.configurations.forEach(configuration => {
                 configurationSnippet.push(configuration.GetLaunchConfiguration(false));
                 configurationSnippet.push(configuration.GetAttachConfiguration());
             });
 
-            util.packageJson.contributes.debuggers[0].configurationSnippets = configurationSnippet.map(snippet => {
-                // Remove internal fields
-                delete snippet.isInitialConfiguration;
-                delete snippet.debuggerType;
-
-                return snippet;
+            configurationSnippet.forEach(snippet => {
+                completionItems.push(convertConfigurationSnippetToCompetionItem(snippet));
```

Comment:
Why don't you just do this once? why do it twice? Am I missing something?
Get rid of configurationSnippet and loop the foreach around this.configurations?